### PR TITLE
Add initial v1alpha1 rate limit API

### PR DESCRIPTION
**Pull request by marcogschmidt**
Manually copied over the `v0.1.0` version of the `v1alpha1` [rate limit API](https://github.com/solo-io/rate-limiter/blob/master/api/v1alpha1/ratelimit.proto) for now. When the new rate limit API is done I will come back too add automation as described in [this design doc](https://github.com/solo-io/dev-docs/pull/26/files) (see the _Publish API_ section).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.PHONY: generate
+generate: generate-code generate-mocks tidy
+
+.PHONY: generate-code
+generate-code:
+	rm -rf vendor_any || true
```
Comment (kdorosh) [inline Makefile:6]: no need for the `|| true`

File: codegen/rate_limiter.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+func rateLimiterGroups() []model.Group {
+	return []model.Group{
+		{
+			GroupVersion: schema.GroupVersion{
+				Group:   "ratelimit.solo.io",
+				Version: "v1alpha1",
+			},
+			Module: "github.com/solo-io/solo-apis", // TODO
```
Comment (kdorosh) [inline codegen/rate_limiter.go:20]: why the TODO?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module github.com/solo-io/solo-apis
+
+go 1.14
+
+replace (
+	github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309
+	k8s.io/client-go => k8s.io/client-go v0.17.1
```
Comment (kdorosh) [inline go.mod:7]: should we make this 0.18 to match `k8s.io/apimachinery`?

Comment (EItanya) [reply go.mod:7]: Everything should be `0.17`. `0.18` has breaking changes to the client functions.

File: codegen/rate_limiter.go (hunk 2 above)
Comment (marcogschmidt) [reply codegen/rate_limiter.go:20]: Forgot to remove it, it was a personal note to check how the field is used.

File: go.mod (hunk 3 above)
Comment (marcogschmidt) [reply go.mod:7]: I actually just fixed this locally. I am actually pinning `apimachinery` to `0.17.1` as well, as that is the version of the k8s libraries we use in skv2.

**Comment by marcogschmidt** [comment]
Added WIP label as this needs an skv2 change.

**Pull request merged**